verilog/tools/syntax/README.md

```diff
@@ -188,6 +193,68 @@ directly to a named grammar construct in the [SV-LRM]. The
+
+| Key         | Type   | Description                                              |
+|-------------|--------|----------------------------------------------------------|
+| `tokens`    | array  | List of [Token](#Token-object) objects, with whitespace tokens filtered out. Present only when `--printtokens` flag is specified. |
+| `rawtokens` | array  | List of [Token](#Token-object) objects. Present only when `--printrawtokens` flag is specified. |
+| `tree`      | object | [Parser tree](#Parser-tree). Present only when `--printtree` flag is specified and parsing errors didn't prevent tree creation. |
+| `errors`    | array  | List of [Error](#Error-object) objects. Present only when there were any errors. |
+
+#### Parser tree
+
+The tree consist of [Node](#Node-object) and [Token](#Token-object) objects. The tree root is a Node object.
+
+#### Node object
+
+| Key        | Type   | Description                                            |
+|------------|--------|--------------------------------------------------------|
+| `tag`      | string | Node tag. See `NodeEnum` in [verilog\_nonterminals.h](../../CST/verilog_nonterminals.h) for available values. |
+| `children` | array  | List of children ([Node](#Node-object) and [Token](#Token-object), or `null`). |
+
+#### Token object
+
+| Key            | Type   | Description                                        |
+|----------------|--------|----------------------------------------------------|
+| `start`, `end` | int    | Byte offset of token's first character and a character just past the symbol in source text. |
+| `tag`          | string | Token tag. See [Possible token tag values](#possible-token-tag-values) below for details. |
+
+To get token text, read source file from byte `start` (included) to byte `end` (excluded). Example in Python:
+
+```python
+start = token["start"]
+end = token["end"]
+
+# Read source file contents as bytes
+with open(source_file_path, "rb") as f:
+    source = f.read()
+
+# Get token text from source file contents
+text = source[start:end].decode("utf-8")
+```
+
+##### Possible token `tag` values
+
+Token tag enumerations come from the [parser generator](../../parser/verilog.y). There are 2 types of values:
+
+* Named tokens (e.g. `SymbolIdentifier`, `TK_DecNumber`), which come from `%token TOKEN_TAG` lines.
+* Single character tokens (e.g. `;`, `=`). They can be found using `'.'` regular expression.
+
+#### Error object
+
+| Key              | Type   | Description                                      |
+|------------------|--------|--------------------------------------------------|
+| `line`, `column` | int    | Line and column in source text. 0-based.         |
+| `text`           | string | Character sequence which caused the error.       |
+| `phase`          | string | Phase during which the error occured. One of: `lex`, `parse`, `preprocess`, `unknown`. |
+| `message`        | string | (optional) Error explanation.                    |
+
```

Comment:
Did you mean the toplevel readme @fangism ? I think it would make sense to have a sentence talking about
`verible-verilog-syntax` in the toplevel readme that there is an JSON output and there are Python examples. This will get people excited enough to check out this specific README.